pkg/profiler/cpu/bpf/maps/maps.go

```diff
@@ -1485,6 +1505,82 @@ func (m *Maps) RefreshProcessInfo(pid int, shouldUseFPByDefault bool) {
 	}
 }
 
+func (m *Maps) GetUnwindFailedReasons() (map[int]profiler.UnwindFailedReasons, error) {
+	ret := make(map[int]profiler.UnwindFailedReasons)
+	it := m.unwindFailedReasons.Iterator()
+	for it.Next() {
+		key := it.Key()
+		var pid int32
+		if err := binary.Read(bytes.NewBuffer(key), m.byteOrder, &pid); err != nil {
+			return nil, err
+		}
+		val, err := m.unwindFailedReasons.GetValue(unsafe.Pointer(&key[0]))
+		if err != nil {
+			return nil, err
+		}
+		var reasons profiler.UnwindFailedReasons
+		if err := binary.Read(bytes.NewBuffer(val), m.byteOrder, &reasons.PcNotCovered); err != nil {
```

Comment:
I'd just create one reader and use it for all calls to binary.Read.  Also I'd probably use reflection in a loop to set the values instead of hand writing each one.

Reply:
I didn't know Go supported reflection, that is definitely way nicer than what I have here (and also the string generation stuff in main.go). Will restructure.